src/aleph_client/commands/instance/__init__.py

```diff
@@ -770,16 +842,60 @@ async def confidential_start(
     valid = await client.validate_measure(sev_data, tik_path, firmware_hash=firmware_hash)
     if not valid:
         echo("Could not validate authenticity of the VM. Please check that you are validating against the proper hash")
-        return 2
+        return 1
     echo("Measurement are authentic")
 
-    secret_key = Prompt.ask(
-        "Please enter secret to start the VM",
-    )
+    secret_key = vm_secret or Prompt.ask("Please enter secret to start the VM", password=True)
 
     encoded_packet_header, encoded_secret = await client.build_secret(tek_path, tik_path, sev_data, secret_key)
     await client.inject_secret(vm_hash, encoded_packet_header, encoded_secret)
-    echo("Starting the instance...")
-    echo("Logs can be fetched using the `aleph instance logs` command")
-    echo("Run the following command to get the IPv6 address of your instance:  aleph instance list")
     await client.close()
+    console = Console()
+    console.print(
+        "Your instance is currently starting..." "\n\nLogs can be fetched using:\n\n",
+        Text.assemble(
+            "  aleph instance logs ",
+            Text(vm_id, style="bright_cyan"),
+            " ",
+            Text(domain, style="blue"),
+            style="italic",
+        ),
+        "\n\nTo get the IPv6 address of the instance, check out:\n\n",
+        Text.assemble(
+            "  aleph instance list\n",
+            style="italic",
+        ),
+    )
+
+
+@app.command()
+async def confidential(
+    vm_id: str,
+    domain: str,
+    policy: int = typer.Option(default=0x1),
+    firmware_hash: str = typer.Option(
+        settings.DEFAULT_CONFIDENTIAL_FIRMWARE_HASH, help=help_strings.CONFIDENTIAL_FIRMWARE_HASH
+    ),
+    firmware_file: str = typer.Option(None, help=help_strings.PRIVATE_KEY),
+    keep_session: bool = typer.Option(None, help=help_strings.KEEP_SESSION),
+    vm_secret: str = typer.Option(None, help=help_strings.VM_SECRET),
+    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
+    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
+    debug: bool = False,
+):
+    "After creating a confidential instance, all-in-one command to initialize, validate and start the VM (allocate + confidential-init-session + confidential-start)"
+    allocated = (await allocate(vm_id, domain, private_key, private_key_file, debug)) is None
```

Comment:
In the normal case allocation is not needed, as we call allocate in the create step.

Reply:
It was just in case, because in some situations the call was failing. I changed the process in the new commit.